Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -47,6 +47,9 @@ class TIFFLoadingContext {
         if (m_metadata.strip_offsets()->size() != m_metadata.strip_byte_counts()->size())
             return Error::from_string_literal("TIFFImageDecoderPlugin: StripsOffset and StripByteCount have different sizes, aborting...");
 
+        if (any_of(*m_metadata.bits_per_sample(), [](auto bit_depth) { return bit_depth == 0 || bit_depth > 32; }))
+            return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid value in BitsPerSample, aborting...");
```

Comment:
Does tiff allow 32 bits per sample? I vaguely remember it tops out at 16, but maybe I'm misremembering.

Reply:
We already had this conversation :laughing:
https://github.com/SerenityOS/serenity/pull/22223#discussion_r1421802741

I'm also quite sure it's written somewhere that for one of the subformat it shouldn't be above 16, but it's not clear to me that there is a general requirement on that.
Also, some of the image in libtiff's test suite have a 32 bit depth.